scaaml/io/dataset.py

```diff
@@ -675,41 +677,41 @@ def _check_shard_metadata(shard_info: Dict, dataset_path: Path) -> None:
+            "path",  # Checked by Dataset._check_sha256sums
+            "group",  # Checked against path
+            "key",  # Checked against path
+            "part",  # Checked against path
+            "size",  # Checked here
+            "chip_id",  # Checked that it is a non-negative integer
         }
         if set(shard_info.keys()) != si_keys:
-            raise ValueError(f'Shard info keys are: {shard_info.keys()} '
-                             f'expected: {si_keys}, in shard: {shard_info}')
+            raise ValueError(f"Shard info keys are: {shard_info.keys()} "
+                             f"expected: {si_keys}, in shard: {shard_info}")
         # Check that the info corresponds to the filename:
-        file_info = Dataset._shard_info_from_name(shard_info['path'])
-        for key in ['group', 'part']:
-            if file_info['shard_' + key] != shard_info[key]:
-                raise ValueError(f'{key} does not match filename, expected: '
-                                 f'{file_info["shard_" + key]}, got: '
-                                 f'{shard_info[key]}, in shard: {shard_info}')
+        file_info = Dataset._shard_info_from_name(shard_info["path"])
+        for key in ["group", "part"]:
+            if file_info["shard_" + key] != shard_info[key]:
+                raise ValueError(f"{key} does not match filename, expected: "
+                                 f"{file_info['shard_' + key]}, got: "
+                                 f"{shard_info[key]}, in shard: {shard_info}")
         # Check key (in filename it is lower case, in info it is upper case)
-        if file_info['shard_key'].lower() != shard_info['key'].lower():
-            raise ValueError(f'key does not match filename, expected: '
-                             f'{file_info["shard_key"]}, got: '
-                             f'{shard_info["key"]} (not case sensitive), in '
-                             f'shard: {shard_info}')
+        if file_info["shard_key"].lower() != shard_info["key"].lower():
+            raise ValueError(f"key does not match filename, expected: "
+                             f"{file_info['shard_key']}, got: "
+                             f"{shard_info['key']} (not case sensitive), in "
+                             f"shard: {shard_info}")
         # Check size of the file
-        size = os.stat(dataset_path / shard_info['path']).st_size
-        if size != shard_info['size']:
-            raise ValueError(f'Wrong size, got: {size}, expected: '
-                             f'{shard_info["size"]}, in shard: {shard_info}')
+        size = os.stat(dataset_path / shard_info["path"]).st_size
+        if size != shard_info["size"]:
+            raise ValueError(f"Wrong size, got: {size}, expected: "
+                             f"{shard_info['size']}, in shard: {shard_info}")
         # Check chip_id is non-negative integer
-        chip_id = shard_info['chip_id']
+        chip_id = shard_info["chip_id"]
```

Comment:
Reviewing this code, I'm thinking that all the text constants that are dictionary keys should all be declared as a constants in a class instead of repeating them all over the place.
Otherwise any change in the schema will be horrible to do and one typo somewhere can lead to unexpected behaviour